File: test_yahoo_search.py
import requests
import json

# 测试后端的商品比较接口
url = "http://localhost:9090/goods/compare"
headers = {
    "Content-Type": "application/json"
}
data = {
    "keyword": "手机",
    "page": 1
}

print("发送商品比较请求...")
response = requests.post(url, headers=headers, json=data)

print(f"状态码: {response.status_code}")
print(f"响应内容: {response.text}")

if response.status_code == 200:
    result = response.json()
    print(f"返回的商品数量: {len(result.get('goodsList', []))}")
    # 检查是否有雅虎来源的商品
    yahoo_goods = [g for g in result.get('goodsList', []) if g.get('source') == 'yahoo']
    print(f"雅虎商品数量: {len(yahoo_goods)}")
    if yahoo_goods:
        print("雅虎商品示例:")
        for i, goods in enumerate(yahoo_goods[:3]):
            print(f"  {i+1}. {goods.get('goodsName')} - ¥{goods.get('goodsPrice')} - {goods.get('source')}")
    else:
        print("没有找到雅虎来源的商品!")
else:
    print("请求失败!")

File: test_yahoo_search.js
const http = require('http');

// 测试Yahoo搜索API
function testYahooSearch() {
    const postData = JSON.stringify({
        query: '手机',
        pageNum: 1,
        pageSize: 10
    });
    
    const options = {
        hostname: 'localhost',
        port: 9090,
        path: '/goods/search',
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': Buffer.byteLength(postData)
        }
    };

    const req = http.request(options, (res) => {
        let data = '';
        res.on('data', (chunk) => {
            data += chunk;
        });
        res.on('end', () => {
            console.log('🔍 Yahoo搜索测试结果:');
            console.log('状态码:', res.statusCode);
            console.log('响应头:', res.headers);
            try {
                const jsonData = JSON.parse(data);
                console.log('响应数据:', JSON.stringify(jsonData, null, 2));
                if (jsonData.data) {
                    console.log('商品数量:', Array.isArray(jsonData.data) ? jsonData.data.length : 'Not array');
                    if (Array.isArray(jsonData.data)) {
                        jsonData.data.forEach((item, index) => {
                            console.log(`${index + 1}. ${item.goodsName} - ¥${item.goodsPrice} (${item.mallType === 20 ? 'Yahoo' : 'Other'})`);
                        });
                    }
                }
            } catch (e) {
                console.log('原始响应:', data);
                console.log('JSON解析失败:', e.message);
            }
        });
    });

    req.on('error', (error) => {
        console.log('❌ 请求错误:', error.message);
    });

    req.write(postData);
    req.end();
}

// 执行测试
console.log('🚀 开始测试Yahoo搜索功能...');
testYahooSearch();